Dalamud/Game/Addon/Events/AddonEventManager.cs

```diff
@@ -87,6 +92,8 @@ public void Dispose()
     /// <returns>IAddonEventHandle used to remove the event.</returns>
     internal IAddonEventHandle? AddEvent(string pluginId, IntPtr atkUnitBase, IntPtr atkResNode, AddonEventType eventType, IAddonEventManager.AddonEventHandler eventHandler)
     {
+        if (!ThreadSafety.IsMainThread) throw new NotSupportedException("This should be done only from the main thread. Modifying active native code on non-main thread is not supported.");
```

Comment:
Nit: InvalidOperationException would be better here. Will change it really quick.

Reply:
Haha it was originally InvalidOperationException, when I presented my modification to the wording Kizer suggested NotSupportedException instead.